homework/index.js

```diff
@@ -1,47 +1,238 @@
       }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+    });
   }
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
+  function fillRepositoryDetailsTable(selectedRepository) {
+    const tableBody = document.querySelector('#table-body');
+
+    const repository = {
+      repository: selectedRepository.name,
+      description: selectedRepository.description,
+      forks: selectedRepository.forks,
+      updated: new Date(selectedRepository.updated_at).toLocaleString(),
+    };
+
+    Object.keys(repository).forEach((key, index) => {
+      const tr = Helper.createAndAppend('tr', tableBody);
+
+      if (index === 0) {
+        Helper.createAndAppend('td', tr, { class: 'label', text: key });
+        const td = Helper.createAndAppend('td', tr);
+        Helper.createAndAppend('a', td, {
+          href: selectedRepository.html_url,
+          target: '_blank',
+          text: repository[key],
+        });
       } else {
-        elem.setAttribute(key, value);
+        Helper.createAndAppend('td', tr, { class: 'label', text: key });
+        Helper.createAndAppend('td', tr, { text: repository[key] });
       }
     });
-    return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
+  async function fillContributorsList(selectedRepository) {
+    const list = document.querySelector('.contributors__list');
+    try {
+      const contributors = await Helper.fetchJSON(selectedRepository.contributors_url);
+      if (!contributors.length) {
+        Helper.createAndAppend('li', list, {
+          class: 'alert-warning',
+          text: 'There is no contributor for this repository!',
+        });
+        return;
+      }
+
+      contributors.forEach(contributor => {
+        const listItem = Helper.createAndAppend('li', list, {
+          class: 'contributors__item',
+        });
+        const listLink = Helper.createAndAppend('a', listItem, {
+          href: contributor.html_url,
+          target: '_blank',
+          class: 'contributors__link',
+        });
+        Helper.createAndAppend('img', listLink, {
+          class: 'contributors__avatar',
+          src: contributor.avatar_url,
+          alt: contributor.login,
+        });
+        Helper.createAndAppend('span', listLink, {
+          class: 'contributors__name',
+          text: contributor.login,
+        });
+        Helper.createAndAppend('span', listLink, {
+          class: 'contributors__badge',
+          text: contributor.contributions,
+        });
+      });
+    } catch (error) {
+      Helper.createAndAppend('li', list, { text: error.message, class: 'alert-error' });
+    }
+  }
+
+  class App {
+    constructor(select) {
+      this.select = select;
```

Comment:
If you use this in a static method, the this value refers to the class itself rather than an instance of the class. Therefore, `this` and `App` are equivalent. You could change all references to `this` to `App` and the code would still work:

```js
class App {
  static init(data, root) {
    createStaticComponentsFromModel(componentModel, root);

    const repositories = data.sort((a, b) =>
      a.name.toLowerCase().localeCompare(b.name.toLowerCase()),
    );

    App.select = document.querySelector('.header__select');

    repositories.forEach(repository => {
      Helper.createAndAppend('option', App.select, { text: repository.name });
    });

    App.select.addEventListener('change', () => {
      document.querySelector('tbody').innerHTML = '';
      document.querySelector('.contributors__list').innerHTML = '';
      fillRepositoryDetailsTable(repositories[App.select.selectedIndex]);
      fillContributorsList(repositories[App.select.selectedIndex]);
    });

    fillRepositoryDetailsTable(repositories[App.select.selectedIndex]);
    fillContributorsList(repositories[App.select.selectedIndex]);
  }

  static async main(url) {
    const root = document.getElementById('root');

    try {
      const repositories = await Helper.fetchJSON(url);
      App.init(repositories, root);
    } catch (err) {
      Helper.createAndAppend('div', root, { class: 'alert-error', text: err.message });
    }
  }
}
```

Of course, using a static class like this, while your other code still uses traditional functions seems inconsistent. I know that your second PR takes a different approach, so no need to fix this here.